### PR TITLE
chore: use pointer::cast for safer pointer cast

### PR DESCRIPTION
**Pull request by tesuji**

### CODE REVIEW

File: src/utils/raw.rs [hunk 1]
```diff
@@ -206,7 +206,7 @@ fn symlink_junction_inner(target: &Path, junction: &Path) -> io::Result<()> {
         );
 
         let mut data = [0u8; MAXIMUM_REPARSE_DATA_BUFFER_SIZE];
-        let db = data.as_mut_ptr() as *mut REPARSE_MOUNTPOINT_DATA_BUFFER;
+        let db = data.as_mut_ptr().cast::<REPARSE_MOUNTPOINT_DATA_BUFFER>();
         let buf = &mut (*db).ReparseTarget as *mut WCHAR;
```
Comment (kinnison) [inline src/utils/raw.rs:210]: Why not also do this one?

**Review by kinnison — COMMENTED** [review]
You appear to have missed (at least) one cast.  Was that deliberate?

File: src/utils/raw.rs (hunk 1 above)
Comment (tesuji) [reply src/utils/raw.rs:210]: `pointer::cast` can only be applied to pointer type. In this case `&mut (*db).ReparseTarget` is a reference.

Comment (kinnison) [reply src/utils/raw.rs:210]: Aha, I understand.

**Pull request merged**